### DISCUSSION TITLE
Getting Started with Phase Field Guide

### DISCUSSION THREAD

**Discussion by wd15** [usnistgov/pfhub#1416] [General]
Create a getting started with phase field guide on the PFHub wiki.

[After some discussion](https://gitter.im/chimad-phase-field/chat-meeting-november-2018) with @tkphd and @guyer, we think that a step by step guide to getting started with phase field would be useful. We may need to have a broader discussion about educational material on PFHub beyond the scope of this issue. This issue is not about generating more educational material.

Currently there are a number of open issues regarding educational materials on PFHub, #586, #585, #583, #582. All of those issues relate to this one.

My initial thoughts were along the lines of [this](https://stackoverflow.com/questions/1012573/getting-started-with-haskell), which I found extremely useful when learning Haskell. Maybe it would be worth asking a beginner question somewhere like [Physics Stack Exchange](https://physics.stackexchange.com/). The guide would not require us to generate new material ideally, but to give a narrative about what to do in what order. What to code first and what to read first and what not to waste time on.

List of ideas for teaching methods or philosophies:

 - [fast.ai teaching philosophy](https://www.fast.ai/2016/10/08/teaching-philosophy/)
 - [Getting Started with Haskell](https://stackoverflow.com/questions/1012573/getting-started-with-haskell)
 - [Software Carpentry style](http://swcarpentry.github.io/python-novice-gapminder/)

Other sources for phase field material links:

  - Guyer summer school course???
  - [FiPy 1D Diffusion](https://www.ctcms.nist.gov/fipy/examples/diffusion/generated/examples.diffusion.mesh1D.html)
  - [MMSP beginner diffusion](https://github.com/mesoscale/mmsp/tree/develop/examples/beginners_diffusion)
  - [Moose example](http://mooseframework.org/old/wiki/MooseTutorials/IronChromiumDecomposition/)
  - Provatas and Elder, Phase-Field Methods in Materials Science and Engineering
  - Binder, Programming Phase-Field Modeling
  - [Peter Voorhees phase field lectures](https://pages.nist.gov/pfhub/wiki/voorhees-lectures/)

Ideally, the guide would include as much free material as possible and not require purchasing expensive books.

This issue can be opened up to the wider community at some point especially including Mike Tonks and Nick Provatas who both have teaching experience.

**Comment by stvdwtt** [comment]
How I got started phase field modeling (in Katsuyo's group as an undergrad):

First, a student is asked to solve a 1D diffusion problem (explicitly) and verify the result with an analytic solution (doing some convergence tests). That’s in a language of their choice, usually Matlab or Python. Then they write a spinodal decomposition code (explicit once again) and test a few different situations (also in Matlab or Python). Finally, they re-write the spinodal decomposition code in C++ or Fortran.

[a comment by an automated account is not shown]
